lib/ASTGen/Sources/ASTGen/CompilerBuildConfiguration.swift

```diff
@@ -18,9 +18,14 @@ import SwiftSyntax
 
 /// A build configuration that uses the compiler's ASTContext to answer
 /// queries.
-struct CompilerBuildConfiguration: BuildConfiguration {
+final class CompilerBuildConfiguration: BuildConfiguration {
```

Comment:
Could we keep this a struct and construct it directly in the ASTGenVisitor's `activeClause(in:)` instead?

Reply:
Yes. The real problem is that we don't have locations for import paths, so CompilerBuildConfiguration has mutable state for the location. I'm considering expanding BuildConfiguration to have a new "canImport" check that provides more source locations, which makes all of this cleaner.

Reply:
I really, really, really wanted to avoid having syntax nodes or source locations passed in to the `BuildConfiguration` for philosophical reasons, but `canImport` needs them. I'll go make those changes on the swift-syntax side and clean things up before coming back here to make this a `struct` again and drop the mutable state.

Reply:
Just to be clear, I meant keeping the `conditionLoc` on CompilerBuildConfiguration, but keeping it immutable by constructing it when querying the active clause, it doesn't seem like there are any other ASTGen methods that need it to be mutable state on `self`. I agree that sinking it down into `canImport` would probably be cleaner though.

Reply:
I've moved it back to a `struct` for now